lib/Service/ProvisioningService.php

```diff
@@ -0,0 +1,176 @@
+		$userName = $idTokenPayload->{$displaynameAttribute} ?? null;
+		$quotaAttribute = $this->providerService->getSetting($providerId, ProviderService::SETTING_MAPPING_QUOTA, 'quota');
+		$quota = $idTokenPayload->{$quotaAttribute} ?? null;
+
+		$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_UID, $idTokenPayload, $tokenUserId);
+		$this->eventDispatcher->dispatchTyped($event);
+
+		$backendUser = $this->userMapper->getOrCreate($providerId, $event->getValue());
+		$this->logger->debug('User obtained from the OIDC user backend: ' . $backendUser->getUserId());
+
+		$user = $this->userManager->get($backendUser->getUserId());
+		if ($user === null) {
+			return null;
+		}
+
+		// Update displayname
+		if (isset($userName)) {
+			$newDisplayName = mb_substr($userName, 0, 255);
+			$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_DISPLAYNAME, $idTokenPayload, $newDisplayName);
+		} else {
+			$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_DISPLAYNAME, $idTokenPayload);
+		}
+		$this->eventDispatcher->dispatchTyped($event);
+		$this->logger->debug('Displayname mapping event dispatched');
+		if ($event->hasValue()) {
+			$newDisplayName = $event->getValue();
+			if ($newDisplayName != $backendUser->getDisplayName()) {
+				$backendUser->setDisplayName($newDisplayName);
+				$backendUser = $this->userMapper->update($backendUser);
+			}
+		}
+
+		// Update e-mail
+		$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_EMAIL, $idTokenPayload, $email);
+		$this->eventDispatcher->dispatchTyped($event);
+		$this->logger->debug('Email mapping event dispatched');
+		if ($event->hasValue()) {
+			$user->setEMailAddress($event->getValue());
+		}
+
+		$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_QUOTA, $idTokenPayload, $quota);
+		$this->eventDispatcher->dispatchTyped($event);
+		$this->logger->debug('Quota mapping event dispatched');
+		if ($event->hasValue()) {
+			$user->setQuota($event->getValue());
+		}
+
+		// Update groups
+		if ($this->providerService->getSetting($providerId, ProviderService::SETTING_GROUP_PROVISIONING, '0') === '1') {
+			$this->provisionUserGroups($user, $providerId, $idTokenPayload);
+		}
+
+		return $user;
+	}
+
+	public function provisionUserGroups(IUser $user, int $providerId, object $idTokenPayload): void {
+		$groupsAttribute = $this->providerService->getSetting($providerId, ProviderService::SETTING_MAPPING_GROUPS, 'groups');
+		$groupsData = $idTokenPayload->{$groupsAttribute} ?? null;
+
+		$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_GROUPS, $idTokenPayload, json_encode($groupsData));
+		$this->eventDispatcher->dispatchTyped($event);
+		$this->logger->debug('Group mapping event dispatched');
+
+		if ($event->hasValue()) {
+			$groups = json_decode($event->getValue());
+			$userGroups = $this->groupManager->getUserGroups($user);
+			$syncGroups = [];
+
+			foreach ($groups as $k => $v) {
```

Comment:
With group provisioning enabled but when the IdP does not return the desired attributes, the event data is set to null, then we check if `$event->hasValue()` which is true and then we pass null to `foreach`.

It looks like something that could be avoided :grin:.

Reply:
I added a simple null check for `$event->getValue()`